models/seir/model_mod.f90

```diff
@@ -0,0 +1,455 @@
+real(r8)    :: t_infec           = 3.8_r8      ! Infection time (days)
+real(r8)    :: t_recov           = 14.0_r8     ! Recovery period (days)
+real(r8)    :: t_death           = 7.0_r8      ! Time until death (days)
+real(r8)    :: alpha             = 0.007_r8    ! Vaccination rate (per day)
+integer(i8) :: theta             = 12467       ! New birth and new residents (persons per day)       
+real(r8)    :: mu                = 0.000025_r8 ! Natural death rate (persons per day)
+real(r8)    :: sigma             = 0.05_r8     ! Vaccination inefficacy (e.g., 95% efficient)
+real(r8)    :: beta              = 1.36e-9_r8  ! Transmission rate (per day)
+real(r8)    :: kappa             = 0.00308     ! Mortality rate  
+real(r8)    :: delta_t           = 0.04167_r8  ! Model time step; 1/24 := 1 hour
+integer(i8) :: num_pop           = 331996199   ! Population (US)    
+real(r8)    :: pert_size         = 1.0         ! Size of perturbation (lognormal pdf param)
+
+real(r8)    :: gama, delta, lambda, rho 
+
+! Other related model parameters
+real(r8), parameter :: E0 = 1.0_r8   ! Exposed (not yet infected)
+real(r8), parameter :: I0 = 1.0_r8   ! Infected (not yet quarantined)
+real(r8), parameter :: Q0 = 1.0_r8   ! Quarantined (confirmed and infected)
+real(r8), parameter :: R0 = 1.0_r8   ! Recovered
+real(r8), parameter :: D0 = 1.0_r8   ! Dead
+real(r8), parameter :: V0 = 1.0_r8   ! Vaccinated
+real(r8)            :: S0            ! Susceptible
+
+namelist /model_nml/ model_size, time_step_days, time_step_seconds, &
+                     delta_t, num_pop, pert_size, &
+                     t_incub, t_infec, t_recov, t_death, &
+                     alpha, theta, beta, sigma, kappa, mu   
+
+contains
+
+
+!------------------------------------------------------------------
+!
+! Called to do one time initialization of the model. As examples,
+! might define information about the model size or model timestep.
+! In models that require pre-computed static data, for instance
+! spherical harmonic weights, these would also be computed here.
+! Can be a NULL INTERFACE for the simplest models.
+
+subroutine static_init_model()
+
+real(r8) :: x_loc
+integer  :: i, dom_id
+
+! Do any initial setup needed, including reading the namelist values
+call initialize()
+
+! Create storage for locations
+allocate(state_loc(model_size))
+
+! Define the locations of the model state variables
+! The SEIR variables have no physical location, 
+! and so I'm placing all 7 variables at the same 
+! virtual point in space.
+x_loc = 0.5_r8  
+do i = 1, model_size
+   state_loc(i) =  set_location(x_loc)
+end do
+
+! This time is both the minimum time you can ask the model to advance
+! (for models that can be advanced by filter) and it sets the assimilation
+! window.  All observations within +/- 1/2 this interval from the current
+! model time will be assimilated. If this isn't settable at runtime 
+! feel free to hardcode it and not add it to a namelist.
```

Comment:
comment for SEIR rather than any model

```suggestion
! This time is both the minimum time you can ask the model to advance
! and it sets the assimilation window. 
! All observations within +/- 1/2 this interval from the current
! model time will be assimilated.
```